ale_linters/javascript/flow.vim

```diff
@@ -0,0 +1,66 @@
+" Author: Zach Perrault -- @zperrault
+" Description: FlowType checking for JavaScript files
+
+let g:ale_javascript_flow_executable =
+\   get(g:, 'ale_javascript_flow_executable', 'flow')
+
+let g:ale_javascript_flow_use_global =
+\   get(g:, 'ale_javascript_flow_use_global', 0)
+
+function! ale_linters#javascript#flow#GetExecutable(buffer) abort
+  if g:ale_javascript_flow_use_global
+    return g:ale_javascript_flow_use_global
+  endif
+
+  return ale#util#ResolveLocalPath(
+  \   a:buffer,
+  \   'node_modules/.bin/flow',
+  \   g:ale_javascript_flow_executable
+  \)
+endfunction
+
+function! ale_linters#javascript#flow#GetCommand(buffer) abort
+  return ale_linters#javascript#flow#GetExecutable(a:buffer)
+  \   . ' check-contents --json --from ale'
+endfunction
+
+function! ale_linters#javascript#flow#Handle(buffer, lines)
+  let l:flow_output = json_decode(join(a:lines, ''))
+
+  if l:flow_output
+    let l:output = []
+
+    for l:error in l:flow_output.errors
+      " Each error is broken up into parts
+      let l:text = ''
+      let l:line = 0
+      for l:message in l:error.message
+        " Comments have no line of column information
+        if l:message.line + 0
+          let l:line = l:message.line + 0
+        endif
+        let l:text = l:text . ' ' . l:message.descr
+      endfor
```

Comment:
@w0rp Yeah I was playing with that. The tricky part is that the `Comment` type messages don't have any loc information. 😕